### ISSUE TITLE
When I tried to delete the first <span contenteditable='false'> tag of a block, it raised an error

### ISSUE THREAD

**Issue by amoydavid** [codex-team/editor.js#2852]
I'm using a [editorjs-mention-tool](https://github.com/kreudev/editorjs-mention-tool) plugin.

It will insert a `<span contenteditable='false'>@user</span>` tag into a block inner html.

But if this span tag located at the first element of the block, when I tried to press the `backspace` keyboard button to delete the whole span, I got this error.

```
Failed to execute 'setEnd' on 'Range': There is no child at offset 15.
IndexSizeError: Failed to execute 'setEnd' on 'Range': There is no child at offset 15.
```

situation:

```
<div class="ce-paragraph cdx-block" contenteditable="true" data-placeholder-active="Let's write an awesome story!" data-empty="false"><span href="kyleockford" contenteditable="false" class="mention_tool_at">[[kyleockford]]</span>|</div>
```

My caret is at the place of `|` , then I press backspace and I got the error.

related to this function?
https://github.com/codex-team/editor.js/blob/631340912e7a1b3db1e114e930ddf1e97ec4ea61/src/components/utils/caret.ts#L10

**Comment by neSpecc** [comment]
Yes, that is probably because browser returns incorrect selection `focusOffset`. We need to find a solution.

**Comment by amoydavid** [comment]
@neSpecc I've tried to make a pr #2853  for this. It seems fixed my bug.
